File: 프로그래머스/unrated/181929. 원소들의 곱과 합/원소들의 곱과 합.cs
using System;
using System.Linq;

public class Solution {
    public int solution(int[] num_list) 
    {
        int a = num_list.Aggregate((cur, next) => cur * next);
        
        int sum = num_list.Sum();
        int b = sum * sum;
        
        return a < b ? 1 : 0;
    }
}

File: 프로그래머스/0/181929. 원소들의 곱과 합/원소들의 곱과 합.cpp
#include <string>
#include <vector>
#include <numeric>
#include <cmath>

using namespace std;

int solution(vector<int> num_list) {
    int answer = 0;
    
    int mult = std::accumulate(num_list.begin(), num_list.end(), 1, std::multiplies<int>());
    int sum = std::accumulate(num_list.begin(), num_list.end(), 0);
    int potSum = std::pow(sum, 2);
    
    return mult < potSum ? 1 : 0;
}